### PR TITLE
#11622: add concat sweep

### PR DESCRIPTION
**Pull request by sjameelTT**
### Ticket
#11622

### Problem description
TMs need to be swept. Starting with concat as an example.

### What's changed
Describe the approach used to solve the problem.
Summarize the changes made and its impact.

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tests/sweep_framework/sweeps/data_movement/concat/concat_sharded.py [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+                        ttnn.CoreCoord(0, 0),
+                        ttnn.CoreCoord(concat_specs["num_cores1"][1] - 1, concat_specs["num_cores1"][0] - 1),
+                    )
+                }
+            ),
+            (concat_specs["core_shape1"][0], concat_specs["core_shape1"][1]),
+            ttnn.ShardOrientation.ROW_MAJOR,
+            False,
+        )
+        input_b_shard_spec = ttnn.ShardSpec(
```
Comment (tarafdarTT) [inline tests/sweep_framework/sweeps/data_movement/concat/concat_sharded.py:302]: input_b_shard_spec = ttnn.create_sharded_memory_config(
        shape=(concat_specs["shape2"][0], concat_specs["shape2"][0]),
        core_grid=(concat_specs["num_cores2"][1] - 1,  concat_specs["num_cores2"][0] - 1),
        strategy=ttnn.ShardStrategy.HEIGHT,
        orientation=ttnn.ShardOrientation.ROW_MAJOR,
        use_height_and_width_as_shard_shape=False,
    )

File: tests/sweep_framework/sweeps/data_movement/concat/concat_sharded.py [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+                    )
+                }
+            ),
+            (concat_specs["core_shape2"][0], concat_specs["core_shape2"][1]),
+            ttnn.ShardOrientation.ROW_MAJOR,
+            False,
+        )
+    else:
+        input_a_shard_spec = ttnn.ShardSpec(
+            ttnn.CoreRangeSet(
```
Comment (tarafdarTT) [inline tests/sweep_framework/sweeps/data_movement/concat/concat_sharded.py:317]: input_a_shard_spec = ttnn.create_sharded_memory_config(
        shape=(concat_specs["shape1"][0], concat_specs["shape1"][0]),
        core_grid=(concat_specs["num_cores1"][1] - 1,  concat_specs["num_cores1"][0] - 1),
        strategy=ttnn.ShardStrategy.WIDTH,
        orientation=ttnn.ShardOrientation.ROW_MAJOR,
        use_height_and_width_as_shard_shape=False,
    )

File: tests/sweep_framework/sweeps/data_movement/concat/concat_sharded.py [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+                        ttnn.CoreCoord(concat_specs["num_cores2"][1] - 1, concat_specs["num_cores2"][0] - 1),
+                    )
+                }
+            ),
+            (concat_specs["core_shape2"][0], concat_specs["core_shape2"][1]),
+            ttnn.ShardOrientation.ROW_MAJOR,
+            False,
+        )
+    elif input_mem_config == ttnn.L1_HEIGHT_SHARDED_MEMORY_CONFIG:
+        input_a_shard_spec = ttnn.ShardSpec(
```
[the remark this answers is not in the document]
Comment (sjameelTT) [reply tests/sweep_framework/sweeps/data_movement/concat/concat_sharded.py]: Done

File: tests/sweep_framework/sweeps/data_movement/concat/concat_sharded.py (hunk 1 above)
Comment (sjameelTT) [reply tests/sweep_framework/sweeps/data_movement/concat/concat_sharded.py]: Done

File: tests/sweep_framework/sweeps/data_movement/concat/concat_sharded.py (hunk 2 above)
Comment (sjameelTT) [reply tests/sweep_framework/sweeps/data_movement/concat/concat_sharded.py]: Done

File: tests/sweep_framework/sweeps/data_movement/concat/concat_sharded.py [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+                        ttnn.CoreCoord(0, 0),
+                        ttnn.CoreCoord(concat_specs["num_cores1"][1] - 1, concat_specs["num_cores1"][0] - 1),
+                    )
+                }
+            ),
+            (concat_specs["core_shape1"][0], concat_specs["core_shape1"][1]),
+            ttnn.ShardOrientation.ROW_MAJOR,
+            False,
+        )
+        input_b_shard_spec = ttnn.ShardSpec(
```
[the remark this answers is not in the document]
Comment (sjameelTT) [reply tests/sweep_framework/sweeps/data_movement/concat/concat_sharded.py]: Done

**Comment by sjameelTT** [comment]
https://github.com/tenstorrent/tt-metal/actions/runs/10584767597

**Review by jdesousa-TT — APPROVED** [review]

**Review by xanderchin — APPROVED** [review]
May want to bump down the timeout if possible? Is 45s necessary right now?

**Comment by sjameelTT** [comment]
> May want to bump down the timeout if possible? Is 45s necessary right now?

Locally it was a problem because for WH the default reset didn't retrain the ethernet and dram, so the next test would sit and wait for the retraining, causing a situation where you could suspend the tt-smi reset and attempt a reset. This fried my machine last week, so I had that for safety. With the custom tt-smi command though it isn't a problem, so I'll bump it down to 20.

**Review by ntarafdar — APPROVED** [review]

**Pull request merged**